Model/Resource/Storage/WeeeStorage.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace BigBridge\ProductImport\Model\Resource\Storage;
+
+use BigBridge\ProductImport\Api\Data\Product;
+use BigBridge\ProductImport\Model\Data\EavAttributeInfo;
+use BigBridge\ProductImport\Model\Persistence\Magento2DbConnection;
+use BigBridge\ProductImport\Model\Resource\MetaData;
+
+class WeeeStorage
+{
+    /** @var Magento2DbConnection */
+    protected $db;
+
+    /** @var MetaData */
+    protected $metaData;
+
+    /**
+     * @param Magento2DbConnection $db
+     * @param MetaData $metaData
+     */
+    public function __construct(Magento2DbConnection $db, MetaData $metaData)
+    {
+        $this->db = $db;
+        $this->metaData = $metaData;
+    }
+
+    /**
+     * @param Product[] $products
+     */
+    public function updateWeees(array $products)
+    {
+        $affectedProducts = [];
+
+        foreach ($products as $product) {
+            if ($product->getWeees() !== null) {
+                $affectedProducts[] = $product;
+            }
+        }
+
+        $this->removeWeees($affectedProducts);
+        $this->insertWeees($affectedProducts);
+    }
+
+    /**
+     * @param Product[] $products
+     */
+    public function insertWeees(array $products)
+    {
+        foreach ($products as $product) {
+
+            foreach ($weees = $product->getWeees() as $i => $weee) {
+
+                $this->db->execute("
+                    INSERT INTO `{$this->metaData->weeeTable}`
+                    SET
+                        `website_id` = ?,
+                        `entity_id` = ?,
+                        `country` = ?,
+                        `value` = ?,
+                        `state` = ?,
+                        `attribute_id` = ?
+                ", [
+                    (int)$weee->getWebsiteId(),
+                    $product->id,
+                    $weee->getCountry(),
+                    $weee->getValue(),
+                    (int)$weee->getState(),
+                    $product->getWeeeId()
```

Comment:
Hi!

First of all: this is awesome! Thank you very much for improving the importer. I hadn't even thought about adding weee. And your code changes look really good. You followed my idioms really well. Thanks!

I understand your change assumes that there is only one weee attribute. There can be multiple, of course. But I don't mind much about this. If someone needs more, which I doubt, we will see about that later.

But I was wondering about why you store the weee attribute id in the product. I think you can do without, because the attribute id is already present in the MetaData, and you have the MetaData object present at this point in the code. Is it possible that you remove Product::setWeeeId() and getWeeeId()? If possible then this could remove a point of confusion for the other users of the library. WeeeResolver::resolveWeeeAttributeId() may still do the presence check, but not return an id.

Patrick

Reply:
Hi!

Thanks for your words. I'm glad you like my feature and thank you for create this extension that saves us much time.

I've followed your suggestion, you were totally right, and I've uploaded a new commit with the changes.

Jessica